src/metrics/not_covered_frames.py

```diff
@@ -0,0 +1,65 @@
+import copy
+
+from nptyping import Float, NDArray, Shape
+from src.core import Database, PointCloud
+from src.metrics.reduction_metric import ReductionMetric
+from tqdm.contrib import tzip
+
+
+class NotCoveredFrames(ReductionMetric):
+    """
+    The metric returns number of frames not covered by the database
+    """
+
+    def __init__(self, threshold: float, voxel_size: float = 0.1):
+        """
+        Constructs CoverageDiscarded reduction metric
+        :param voxel_size: Voxel size for down sampling
+        :param threshold: Coverage threshold for the frame to be considered uncovered
+        """
+        self.threshold = threshold
+        self.voxel_size = voxel_size
+
+    def evaluate(self, original_db: Database, filtered_db: Database) -> int:
+        bounds = original_db.get_bounds()
+        result = self._evaluate_with_bounds(original_db, filtered_db, bounds)
+        return result
+
+    evaluate.__doc__ = ReductionMetric.evaluate.__doc__
+
+    def evaluate_many(
+        self, original_db: Database, filtered_dbs: list[Database]
+    ) -> list[int]:
+        bounds = original_db.get_bounds()
+
+        results = []
+        for filtered_db in filtered_dbs:
+            results.append(self._evaluate_with_bounds(original_db, filtered_db, bounds))
```

Comment:
Don't like this. Results of map point cloud should be cached somewhere to avoid this